components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/test/java/org/wso2/carbon/identity/application/authentication/framework/util/FrameworkUtilsTest.java

```diff
@@ -188,4 +379,125 @@ public void testBuildURLWithQueryParams(String url, Map<String, String> queryPar
+
+        assertEquals(removedCookie.getName(), FrameworkConstants.COMMONAUTH_COOKIE);
+        assertEquals(removedCookie.getPath(), ROOT_DOMAIN);
+        assertEquals(removedCookie.getMaxAge(), 0);
+    }
+
+    @Test
+    public void testRemoveCookieExistCookieConfig() {
+        mockCookieTest();
+
+        IdentityCookieConfig cookieConfig = new IdentityCookieConfig(FrameworkConstants.COMMONAUTH_COOKIE);
+        mockStatic(IdentityUtil.class);
+        when(IdentityUtil.getIdentityCookieConfig(FrameworkConstants.COMMONAUTH_COOKIE)).thenReturn(cookieConfig);
+
+        FrameworkUtils.removeCookie(request, response, FrameworkConstants.COMMONAUTH_COOKIE,
+                SameSiteCookie.STRICT, ROOT_DOMAIN);
+
+        verify(response, times(1)).addCookie(cookieCaptor.capture());
+        List<Cookie> capturedCookies = cookieCaptor.getAllValues();
+        Cookie removedCookie = capturedCookies.get(0);
+
+        assertEquals(removedCookie.getName(), FrameworkConstants.COMMONAUTH_COOKIE);
+        assertEquals(removedCookie.getPath(), ROOT_DOMAIN);
+        assertEquals(removedCookie.getMaxAge(), 0);
+
+    }
+
+    @Test
+    public void testRemoveCookieNonExistCookie() {
+        mockCookieTest();
+
+        FrameworkUtils.removeCookie(request, response, "NonExistingCookie",
+                SameSiteCookie.STRICT, ROOT_DOMAIN);
+
+        verify(response, never()).addCookie(anyObject());
+    }
+
+    @Test
+    public void testSetCookieNonExistCookieConfig() {
+        mockCookieTest();
+        int age = 3600;
+
+        FrameworkUtils.setCookie(request, response, FrameworkConstants.COMMONAUTH_COOKIE, "commonAuthIdValue", age);
+
+        verify(response, times(1)).addCookie(cookieCaptor.capture());
+        List<Cookie> capturedCookies = cookieCaptor.getAllValues();
+        Cookie storedCookie = capturedCookies.get(0);
+
+        assertEquals(storedCookie.getName(), FrameworkConstants.COMMONAUTH_COOKIE);
+        assertEquals(storedCookie.getPath(), ROOT_DOMAIN);
+        assertEquals(storedCookie.getMaxAge(), age);
+
+    }
+
+    @Test
+    public void testSetCookieExistCookieConfig() {
+        mockCookieTest();
+
+        IdentityCookieConfig cookieConfig = new IdentityCookieConfig(FrameworkConstants.COMMONAUTH_COOKIE);
+        mockStatic(IdentityUtil.class);
+        when(IdentityUtil.getIdentityCookieConfig(FrameworkConstants.COMMONAUTH_COOKIE)).thenReturn(cookieConfig);
+        int age = 3600;
+
+        FrameworkUtils.setCookie(request, response, FrameworkConstants.COMMONAUTH_COOKIE, "commonAuthIdValue", age);
+
+        verify(response, times(1)).addCookie(cookieCaptor.capture());
+        List<Cookie> capturedCookies = cookieCaptor.getAllValues();
+        Cookie storedCookie = capturedCookies.get(0);
+
+        assertEquals(storedCookie.getName(), FrameworkConstants.COMMONAUTH_COOKIE);
+        assertEquals(storedCookie.getPath(), ROOT_DOMAIN);
+        assertEquals(storedCookie.getMaxAge(), age);
+    }
+
```

Comment:
Let's remove this new line.

Reply:
Addressed with commit c4e56d45a2072535df672c98b16637ada8cad476 in https://github.com/wso2/carbon-identity-framework/pull/3774 PR